### PR TITLE
Netstat collector - TCP connection states

### PR DESCRIPTION
**Pull request by gjedeer**
This pull request implements a simple collector using /proc/net/tcp to publish number of connections in TCP states.

contributors1

It's https://github.com/BrightcoveOS/Diamond/pull/797 rebased to python-diamond

### CODE REVIEW

**Comment by shortdudey123** [comment]
can you create `docs/collectors/NetstatCollector.md` as well?
Should me able to use `make docs` to help you

**Comment by shortdudey123** [comment]
Also, fix all the pep8 errors :)

**Comment by gjedeer** [comment]
As requested.

**Comment by gjedeer** [comment]
Hmm, Travis pep8 check fails. It worked on my machine with the "pep8" command from pip.

I have no idea what this error means:

    src/collectors/netstat/netstat.py:25:13: E126 continuation line over-indented for hanging indent

**Comment by shortdudey123** [comment]
Verify you are testing w/ pep8 `1.5.7`

Indent line 25 8 spaces and not 12

**Comment by shortdudey123** [comment]
The `cython` error can be ignored.
LGTM

File: src/collectors/netstat/test/testnetstat.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        self.collector = NetstatCollector(config, None)
+
+    @patch.object(Collector, 'publish')
+    def test(self, publish_mock):
+        NetstatCollector.PROC_TCP = self.getFixturePath('proc_net_tcp')
+        self.collector.collect()
+
+        metrics = {
+            'LISTEN':  9
```
Comment (shortdudey123) [inline src/collectors/netstat/test/testnetstat.py]: Can you expand this a bit? there is quite a few more metrics that should end up getting reported

**Comment by shortdudey123** [comment]
Can you rebase on master (cython was fixed so testing will work) and squash your commits?

File: docs/collectors/NetstatCollector.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+byte_unit | byte | Default numeric output(s) | str
+enabled | False | Enable collecting these metrics | bool
+measure_collector_time | False | Collect the collector run time in ms | bool
+metrics_blacklist | None | Regex to match metrics to block. Mutually exclusive with metrics_whitelist | NoneType
+metrics_whitelist | None | Regex to match metrics to transmit. Mutually exclusive with metrics_blacklist | NoneType
+
+#### Example Output
+
+```
+__EXAMPLESHERE__
```
Comment (shortdudey123) [inline docs/collectors/NetstatCollector.md]: Since you have `setDocExample` in the tests, this should end up not being the templated output

**Pull request merged**